sdk/servicebus/service-bus/src/core/receiverHelper.ts

```diff
@@ -25,27 +25,30 @@ export class ReceiverHelper {
    * @returns true if credits were added, false if there is no current receiver instance
    * or `stopReceivingMessages` has been called.
    */
-  public addCredit(credits: number): boolean {
+  addCredit(credits: number): boolean {
     const receiver = this._getCurrentReceiver();
 
-    if (this._isSuspended || receiver == null || !receiver.isOpen()) {
+    if (!this.canReceiveMessages()) {
       return false;
     }
 
-    receiver.addCredit(credits);
+    if (receiver != null) {
```

Comment:
Is this check needed? Won't the `if (!this.canReceiveMessages())` check above take care of this?

Reply:
It is totally redundant _but_ TS doesn't know that it's not null based on the results of calling the function above. :|

Reply:
That is when we use the ! bang bang